lib/hasPermissions.ts

```diff
@@ -203,16 +207,44 @@ export class HasPermissionsDirectiveVisitor<
     const hasPermissionsValidateFunction: ValidateFunction<TContext> = (
       value: unknown,
       _: unknown,
-      __: unknown,
+      container: GraphQLArgument | GraphQLInputObjectType | GraphQLObjectType,
       context: TContext,
       resolverInfo: Record<string, unknown>,
       resolverSource: unknown,
       resolverArgs: Record<string, unknown>,
+      path?: Array<string>,
     ): unknown => {
+      if (
+        location ===
+          (DirectiveLocation.INPUT_FIELD_DEFINITION ||
+            DirectiveLocation.INPUT_OBJECT) &&
+        container instanceof GraphQLInputObjectType &&
+        path &&
+        value === container.getFields()[path[path.length - 1]].defaultValue
+      ) {
+        return value;
+      }
+
+      if (
+        location === DirectiveLocation.ARGUMENT_DEFINITION &&
+        container instanceof GraphQLInputObjectType &&
+        path &&
+        value === container.getFields()[path[path.length - 1]].defaultValue
+      ) {
+        return value;
+      }
+
+      if (
+        location === DirectiveLocation.ARGUMENT_DEFINITION &&
+        value === (container as GraphQLArgument).defaultValue
+      ) {
+        return value;
+      }
+
       if (
         !permissions ||
         !permissions.length ||
-        (isUsedOnInputOrArgument && (value === null || value === undefined))
+        (isUsedOnInputOrArgument && value === undefined)
```

Comment:
why did you remove null from there?

Reply:
I'm sorry for just answer now, I didn't see this comment, but answering that, I removed this null, because now if the user doesn't have permissions to send the field, only the default value should be allowed (not even null should be allowed if null is not the default value), and with that `value === null`, the user can be pass `{ someFilter: null }` and receive some value from the query, even without permission.